### ISSUE TITLE
[NativeAOT] Collecting GC trace using perfview does not work

### ISSUE THREAD

**Issue by jkotas** [dotnet/runtime#101328]
## Repro (.NET 8, Windows x64)

1. `dotnet new console --aot`
2. Enable `<EventSourceSupport>true</EventSourceSupport>` in the .csproj
3. Replace `Program.cs` with:
```csharp
int i = 0;
for (;;)
{
    GC.KeepAlive(new int[1000]);
    Console.WriteLine(i++);
}
```
4. `dotnet publish` and start the published binary (bin\Release\net8.0\win-x64\publish\test.exe)

5. Get perfview from https://github.com/microsoft/perfview/releases
6. Go to Collect / Collect / check "Kernel Base", ".NET" and "GC Collect Only" in Advanced options
7. Hit "Start collection", let it collect for 5-10 seconds, hit "Stop collection"
8. Click on the collected trace, go to Memory / GCStats, and find the GCStat for the native AOT published binary process

## Actual results

GC Stats are blank
```
GC Stats for Process 32640: test
 •CommandLine: C:\repro\test\bin\Release\net8.0\win-x64\publish\test.exe
•Runtime Version: V 0.0.0.0
•CLR Startup Flags: None
•Total CPU Time: 0 msec
•Total GC CPU Time: 0 msec
•Total Allocs : 0.000 MB
•Number of Heaps: -1
•GC CPU MSec/MB Alloc : NaN MSec/MB
•Total GC Pause: 0.0 msec
•% Time paused for Garbage Collection: 0.0%
•% CPU Time spent Garbage Collecting: NaN%
•Max GC Heap Size: 0.000 MB
...
``` 

## Expected results

GC Stats show information about GC process activity - similar to what one sees for non-native AOT published binary
```
GC Stats for Process 19084: test
 •CommandLine: C:\repro\test\bin\Release\net8.0\publish\test.exe
•Runtime Version: V 8.0.424.16909
•CLR Startup Flags: 8388611
•Total CPU Time: 0 msec
•Total GC CPU Time: 0 msec
•Total Allocs : 234.862 MB
•Number of Heaps: 1
•GC CPU MSec/MB Alloc : 0.000 MSec/MB
•Total GC Pause: 9.5 msec
...
```

**Comment by agocke** [comment]
@LakshanF dotnet monitor can grab GC info right? I’m curious what preview is doing differently.

**Comment by jkotas** [comment]
I have tried `dotnet-trace collect --profile gc-collect --duration 00:00:00:10 --name test` with the repro above (on .NET 8) and perfview is able to display the gcstats correctly. It means that the ETW channel does not end up with the exact same events as the EventPipe channel.

I have tried the original repro on current .NET 9 and it works fine. It looks like a bug in .NET 8 that we have fixed in .NET 9. It was likely fixed by https://github.com/dotnet/runtime/pull/93245 or some of the changes to dedup more of the implementation with regular CoreCLR.
